integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/reference/FieldReferenceIT.java

```diff
@@ -0,0 +1,524 @@
+
+	private static final SimpleMappedIndex<IndexBinding> index = SimpleMappedIndex.of( IndexBinding::new );
+
+	private static final DataSet dataSet = new DataSet();
+
+	@BeforeAll
+	static void setup() {
+		setupHelper.start().withIndex( index ).setup();
+
+		BulkIndexer bulkIndexer = index.bulkIndexer();
+		dataSet.contribute( index, bulkIndexer );
+		bulkIndexer.join();
+	}
+
+	@Test
+	void test() {
+		assertThatQuery( index.query()
+				.where( f -> f.bool()
+						.should( f.match().field( IndexedEntity_.localDate ).matching( LocalDate.of( 2024, 1, 1 ) ) )
+						.should( f.match().field( IndexedEntity_.localDate.asString() ).matching( "2024-01-02" ) )
+				) )
+				.hasDocRefHitsAnyOrder( index.typeName(), dataSet.docId( 0 ), dataSet.docId( 1 ) );
+
+		assertThatQuery( index.query()
+				.select( f -> f.field( IndexedEntity_.string ) )
+				.where( f -> f.matchAll() ) )
+				.hasHitsAnyOrder( "string_0", "string_1", "string_2" );
+
+		assertThatQuery( index.query()
+				.select( f -> f.field( IndexedEntity_.stringList ).multi() )
+				.where( f -> f.matchAll() ) )
+				.hasHitsAnyOrder(
+						List.of( "string_list_0_0", "string_list_1_0", "string_list_2_0" ),
+						List.of( "string_list_0_1", "string_list_1_1", "string_list_2_1" ),
+						List.of( "string_list_0_2", "string_list_1_2", "string_list_2_2" )
+				);
+
+
+		assertThatQuery( index.query()
+				.select(
+						f -> f.object( IndexedEntity_.nestedEmbeddedEntity )
+								.from( f.field( IndexedEntity_.nestedEmbeddedEntity.string ),
+										f.field( IndexedEntity_.nestedEmbeddedEntity.someEnum ) )
+								.asList()
+				)
+				.where( f -> f.matchAll() ) )
+				.hasHitsAnyOrder(
+						List.of( "string_n_0", SomeEnum.VALUE_1 ),
+						List.of( "string_n_1", SomeEnum.VALUE_2 ),
+						List.of( "string_n_2", SomeEnum.VALUE_3 )
+				);
+
+		assertThatQuery( index.query()
+				.select(
+						f -> f.field( IndexedEntity_.vector )
+				)
+				.where( f -> f.knn( 10 ).field( IndexedEntity_.vector )
+						.matching( new float[] { 1.0f, 1.0f, 1.0f } ) ) )
+				.hasHitsAnyOrder(
+						new float[] { 0.0f, 0.0f, 0.0f },
+						new float[] { 10.0f, 100.0f, 1000.0f },
+						new float[] { 20.0f, 200.0f, 2000.0f }
+				);
+
+		assertThatQuery( index.query()
+				.where( f -> f.nested( IndexedEntity_.nestedEmbeddedEntity )
+						.add( f.match().field( IndexedEntity_.nestedEmbeddedEntity.someEnum )
+								.matching( SomeEnum.VALUE_1 ) ) ) )
+				.hasDocRefHitsAnyOrder( index.typeName(), dataSet.docId( 0 ) );
+
+		assertThatQuery( index.query()
+				.select( f -> f.field( IndexedEntity_.stringProjectedAsBytes ) )
+				.where( f -> f.matchAll() )
+		).hasHitsAnyOrder(
+				new byte[] { 97, 95, 48 },
+				new byte[] { 97, 95, 50 },
+				new byte[] { 97, 95, 49 }
+		);
+
+		assertThatQuery( index.query()
+				.select( f -> f.field( IndexedEntity_.stringProjectedAsBytes.noConverter() ) )
+				.where( f -> f.matchAll() )
+		).hasHitsAnyOrder(
+				"a_0", "a_1", "a_2"
+		);
```

Comment:
This one (`IndexedEntity_.stringProjectedAsBytes`) has a projection converter that returns bytes, so if we just pass it to the projection it'll work with byte[], while no-converter option will produce the string